### PR TITLE
Reduce CVEs: Update dockerfile and dependencies

### PR DESCRIPTION
**Pull request by jcortejoso**
### Description

Update dockerfiles and npm dependencies. Goal is reducing CVEs and also reduce dependencies/size (for system/image dependencies).

#### Before

```
## Overview

                    │                                 Analyzed Image
────────────────────┼──────────────────────────────────────────────────────────────────────────────────
  Target            │  us-west1-docker.pkg.dev/devopsre/social-connect/odis-signer:odis-signer-3.1.1
    digest          │  0fda3ba03742
    platform        │ linux/amd64
    provenance      │ https://github.com/celo-org/social-connect
                    │  1670dc0ec4919e582f287f682dbe7c373d9cd868
    vulnerabilities │    5C    12H    28M    78L     1?
    size            │ 1.2 GB
    packages        │ 2429
                    │
  Base image        │  node:18
                    │  117e3f020487
```

#### After

```
## Overview

                    │           Analyzed Image
────────────────────┼─────────────────────────────────────
  Target            │  signer:latest
    digest          │  c3ebea052be1
    platform        │ linux/amd64
    vulnerabilities │    0C     1H    14M     1L     1?
    size            │ 875 MB
    packages        │ 1889
```

** We need to test on Alfajores **

#### Dependencies updated:

crypto-js: 3.3.0 -> 4.2.0
flat: 4.1.1 -> 5.0.1
protobufjs: 6.11.3/7.2.3 -> 7.2.4
@babel/traverse: 7.21.4 -> 7.23.2
node-fetch: 2.6.0 -> 2.6.7
get-func-name: 2.0.0 -> 2.0.1
minimatch: 3.0.4 -> 3.0.5
async: 2.6.3 -> 2.6.4
browserify-sign: 4.2.1 -> 4.2.2

### CODE REVIEW

**Review by alvarof2 — APPROVED** [review]

**Review by alvarof2 — APPROVED** [review]

File: docs/kubernetes-deployment/signer/values/staging/values-signer0-staging.yaml [hunk 1]
```diff
@@ -46,8 +46,8 @@ env:
 fullnameOverride: ""
 image:
   pullPolicy: Always
-  repository: us-west1-docker.pkg.dev/devopsre/social-connect/odis-signer
-  tag: "odis-signer-3.1.1"
+  repository: us-west1-docker.pkg.dev/devopsre/dev-images/odis-signer
+  tag: latest
```
Comment (soloseng) [inline docs/kubernetes-deployment/signer/values/staging/values-signer0-staging.yaml:50]: I'm a bit reluctant to change this tag to `latest`, as the latest image may not always be the one we use in production.

Could you elaborate on why we would want to use the `latest` tag in the signer deployment yaml?

File: package.json [hunk 2]
```diff
@@ -67,7 +67,16 @@
     "blind-threshold-bls": "npm:@celo/blind-threshold-bls@1.0.0-beta",
     "ganache": "npm:@celo/ganache@7.8.0-unofficial.0",
     "@types/bn.js": "4.11.6",
-    "bignumber.js": "9.0.0"
+    "bignumber.js": "9.0.0",
+    "crypto-js": "4.2.0",
+    "flat": "5.0.1",
+    "protobufjs": "7.2.4",
+    "@babel/traverse": "7.23.2",
+    "node-fetch": "2.6.7",
```
Comment (soloseng) [inline package.json:75]: How were these versions determined? 

Seems that they are not the latest version.

File: docs/kubernetes-deployment/signer/values/staging/values-signer0-staging.yaml (hunk 1 above)
Comment (jcortejoso) [reply docs/kubernetes-deployment/signer/values/staging/values-signer0-staging.yaml:50]: One consideration here is that we're using [policy-controller](https://github.com/sigstore/policy-controller) in all our k8s cluster. One of the things that policy-controller does is, whenever you deploy/upgrade anything (in this case the helm chart for odis releases), it gets the `image` field composed by `<repository>:<tag>`, resolves which image sha corresponds, in that moment, to that tag, and places in k8s the image as `<repository>:<sha>`. That would impact that the deployment would only be upgraded to the `latest` image whenever a `helm upgrade` is done on staging (so it should be always a "controlled" process).

File: package.json (hunk 2 above)
Comment (jcortejoso) [reply package.json:75]: The versions was determined by CVEs reports (using trivy or docker scout). I've just bumped the packages to versions where `CRITICAL` and `MAJOR` CVEs were fixed, but I've not checked if there were more recent versions for each package.

**Pull request merged**